### PR TITLE
feat: Improved adapter and persister registration

### PR DESCRIPTION
**Pull request by offirgolan**
__Adapters__

```js
import { XHRAdapter, FetchAdapter } from '@pollyjs/core';

// Register the xhr adapter so its accessible by all future polly instances
Polly.register(XHRAdapter);

polly.configure({
  adapters: ['xhr', FetchAdapter]
});
```

__Persister__

```js
import { LocalStoragePersister, RESTPersister } from '@pollyjs/core';

// Register the local-storage persister so its accessible by all future polly instances
Polly.register(LocalStoragePersister);

polly.configure({
  persister: 'local-storage'
});

polly.configure({
  persister: RESTPersister
});
```

### CODE REVIEW

File: packages/@pollyjs/core/src/-private/container.js [hunk 1]
```diff
@@ -1,17 +1,78 @@
...
+  }
+
+  /**
+   * Check if a factory has been registered by the
+   * given key (e.g. `adapter:fetch`)
+   *
+   * @param {String} key
+   * @returns {Boolean}
+   */
+  has(key) {
```
Comment (jasonmit) [inline packages/@pollyjs/core/src/-private/container.js]: Should this also take a fn?

File: packages/@pollyjs/core/src/polly.js [hunk 2]
```diff
@@ -119,44 +121,32 @@ export default class Polly {
...
 
-    // Handle Adapters
-    this.config.adapters.forEach(adapter => {
-      let adapterName = adapter;
+    /* Handle Adapters */
 
-      if (isArray(adapterName)) {
-        const [name, AdapterType] = adapterName;
+    // Disconnect from all current adapters
+    this.disconnect();
```
Comment (jasonmit) [inline packages/@pollyjs/core/src/polly.js]: Are there tests asserting disconnect on configure (quick grep didn't find anything)?  Would be useful if there isn't.

File: packages/@pollyjs/core/src/-private/container.js (hunk 1 above)
Comment (jasonmit) [inline packages/@pollyjs/core/src/-private/container.js]: Might be useful for it to take a factory also (but not necessary).

File: packages/@pollyjs/core/src/polly.js (hunk 2 above)
Comment (offirgolan) [reply packages/@pollyjs/core/src/polly.js]: Done.

**Pull request merged**